src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -297,8 +297,9 @@ export class ResultsViewPageStore {
 
             // we get mutations with mutations endpoint, all other alterations with this one, so filter out mutation genetic profile
             const profilesWithoutMutationProfile = _.filter(this.selectedMolecularProfiles.result, (profile: MolecularProfile) => profile.molecularAlterationType !== 'MUTATION_EXTENDED');
+            const genes = this.genes.result;
 
-            if (profilesWithoutMutationProfile.length) {
+            if (profilesWithoutMutationProfile.length && genes != undefined && genes.length) {
```

Comment:
rather than put this check here, would it be possible instead of just not rely on molecularData mobxpromise when we're in geneset mode?  just worth a look because i would rather not introduce into this logic the case where there are no genes, since that is not part of its use case.